Engine/GameEngine.cs

```diff
@@ -0,0 +1,410 @@
+using Engine.Helpers;
+
+namespace Engine;
+
+public static class GameEngine
```

Comment:
I would say this file is probably on the bigger side, maybe there are some constants/methods that could be split out?